src/main/java/org/radarcns/management/domain/MetaToken.java

```diff
@@ -0,0 +1,135 @@
+package org.radarcns.management.domain;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.hibernate.annotations.Cache;
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.envers.Audited;
+import org.radarcns.auth.config.Constants;
+import org.radarcns.management.domain.support.AbstractEntityListener;
+
+@Entity
+@Audited
+@Table(name = "radar_meta_token")
+@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+@EntityListeners({AbstractEntityListener.class})
+public class MetaToken extends AbstractEntity {
+
+
+    //https://math.stackexchange.com/questions/889538/
+    // probability-of-collision-with-randomly-generated-id
+    // Current length of tokenName is 12. If we think there might be collision we can increase
+    // the length.
+    public static final int SHORT_ID_LENGTH = 12;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
+    @SequenceGenerator(name = "sequenceGenerator", initialValue = 1000)
+    private Long id;
+
+    @NotNull
+    @Pattern(regexp = Constants.TOKEN_NAME_REGEX)
+    @Column(name = "token_name", nullable = false, unique = true)
+    private String tokenName;
+
+    @Column(name = "token", length = 2000)
+    private String token;
+
+    @Column(name = "fetched", nullable = false)
+    private Boolean fetched;
+
+    @Column(name = "expiry_date")
+    private ZonedDateTime expiryDate = null;
```

Comment:
Why does this need a time zone? Can it be an `Instant` instead for timezone-independent expiry instant?

Reply:
I made it ZonedDateTime just to be consistant with other time based properties we have in other entities. If Instant is preferred, it can be done

Reply:
For user-local or server-local time notation (e.g. showing something on the screen, logging user input), adding a time zone makes sense. For user-independent time stamps (something should just happen at a certain time) I think Instant is probably less error-prone (because it avoids mapping to the correct time zone).